policies/module-types/template/src/lib.rs

```diff
@@ -0,0 +1,243 @@
+        let output_file = &p.path;
+        let output_file_d = output_file.display();
+
+        // Compute output
+        let output = p
+            .engine
+            .render(p.template_path.as_deref(), p.template_src, p.data)?;
+
+        let already_present = output_file.exists();
+
+        // Check if already correct
+        let mut content = String::new();
+        let already_correct = if already_present {
+            content = read_to_string(output_file)
+                .with_context(|| format!("Failed to read file {output_file_d}"))?;
+            if content == output {
+                true
+            } else {
+                rudder_debug!(
+                    "Output file {} exists but has outdated content",
+                    output_file_d
+                );
+                false
+            }
+        } else {
+            rudder_debug!("Output file {output_file_d} does not exist");
+            false
+        };
+
+        let reported_diff = diff(content, output.clone());
+
+        let outcome = match (already_correct, mode) {
+            (true, _) => Outcome::success(),
+            (false, PolicyMode::Audit) => {
+                if already_present {
+                    bail!(
+                        "Output file {output_file_d} is present but content is not up to date. diff: {reported_diff}"
+                    )
+                } else {
+                    bail!("Output file {output_file_d} does not exist")
+                }
+            }
+            (false, PolicyMode::Enforce) => {
+                // Backup current file
+                if already_present {
+                    backup_file(&output_file, &parameters)?;
+                }
+
+                // Write file
+                fs::write(output_file, output.as_bytes())
+                    .with_context(|| format!("Failed to write file {output_file_d}"))?;
+
+                let source_file = p
+                    .template_path
+                    .as_ref()
+                    .map(|p| format!(" (from {})", p.display()))
+                    .unwrap_or_default();
+
+                if already_present {
+                    Outcome::repaired(format!(
+                        "Replaced {output_file_d} content from template {source_file} diff: {reported_diff}",
+                    ))
+                } else {
+                    Outcome::repaired(format!(
+                        "Written new {output_file_d} from template {source_file} diff: {reported_diff}",
+                    ))
+                }
+            }
+        };
+        Ok(outcome)
+    }
+}
+
+pub fn diff(old: String, new: String) -> String {
+    let diff = TextDiff::from_lines(&old, &new);
+    let mut unified = diff.unified_diff();
+    return unified.context_radius(10).header("old", "new").to_string();
+}
+
+fn backup_file(output_file: &Path, parameters: &Parameters) -> Result<(), anyhow::Error> {
```

Comment:
This function does not need the whole `Parameters` struct, it should probably only take the backip_dir as an `&Path`